src/common/dsp/QuadFilterWaveshaper.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include "globals.h"
+
+const int n_waveshaper_registers = 4;
+
+/*
+ * These are defined in QuadFilterWaveshapers.cpp
+ */
+struct alignas(16) QuadFilterWaveshaperState
+{
+    __m128 R[n_waveshaper_registers];
+    __m128 init;
+};
+typedef __m128 (*WaveshaperQFPtr)(QuadFilterWaveshaperState *__restrict, __m128 in, __m128 drive);
+WaveshaperQFPtr GetQFPtrWaveshaper(int type);
+/*
+ * Given the very first sample inbound to a new voice session, return the
+ * first set of registers for that voice.
+ */
+void initializeWaveshaperRegister(int type, float R[n_waveshaper_registers]);
```

Comment:
Good change! But maybe we should do another step of putting the waveshapers in the filters module now I think of it? But we can do thst separately

Reply:
I wasn't sure if the waveshapers should go into the filters submodule, or maybe they belong in their own submodule? From what I could tell in extracting the filters code, they don't seem to be particularly connected.

Reply:
Yeah we could make them separate true! Submodules are easy enough

Reply:
+1 on having waveshapers in a separate submodule!

Reply:
Ok!